config/config.go

```diff
@@ -45,6 +47,15 @@ type terragruntConfigFile struct {
 	PreventDestroy *bool                  `hcl:"prevent_destroy,attr"`
 	Skip           *bool                  `hcl:"skip,attr"`
 	IamRole        *string                `hcl:"iam_role,attr"`
+
+	// This should be ignored
```

Comment:
Why ignored?

Reply:
This needs to be skipped here because `hcl` can't decode it correctly when it self references `locals`. It is not necessary to decode it into this struct here because it is evaluated in a separate loop.

However, we don't want to completely omit the existence of it because we want a syntax error if we have blocks that terragrunt doesn't expect (e.g typos). Hence it is added here as an attribute but the struct won't attempt to parse anything in it.

Reply:
Perhaps expand the comment a bit to explain this?

Reply:
Already did: https://github.com/gruntwork-io/terragrunt/pull/802/files#diff-3baf47c64847a8fb8aaa8cc2e088513bR51